### PR TITLE
fix: changes in support of clustering

### PR DESCRIPTION
**Pull request by JoeHCQ1**
**Update Sept 23, 2024**:

I'm changing this PR so I can merge the improvements made in the service of clustering even while clustering is not yet fully working.

### CODE REVIEW

File: docs/clustering.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+1. The confluence nodes (pods) want to speak to each other in terms of bare IP addresses instead of pod-names, which are DNS host names resolvable in-cluster. Istio does not "like" pods doing this, and it is suspected to be a
+contributing factor to the inability of the clustering to work with Istio. Note, Hazelcast as built into Confluence is using K8s API based node discovery. 
+[This is one of two Hazelcast supported methods](https://docs.hazelcast.com/hazelcast/latest/kubernetes/kubernetes-auto-discovery#discovering-members). It could also find other nodes by resolving their DNS pod names to an IP. If in
+this mode Hazelcast would be sending messages to `confluence-0` instead of `10.42.0.36`, it might play better with Istio. Injecting Hazelcast settings to alter the node discovery method and/or further work on Istio Destination
+Rules are offered as two approaches that might lead to success.
+
+This url may also help resolve the pod-IP access problem: <https://discuss.istio.io/t/istio-mtls-and-pod-ip-port/7537>.
+
+### Clustering w/o Istio
```
Comment (corang) [inline docs/clustering.md:41]: I'm not sure this is something we should support / document at all to be honest

File: src/namespace/confluence-ns.yaml [hunk 2]
```diff
@@ -1,4 +1,18 @@
+---
```
Comment (corang) [inline src/namespace/confluence-ns.yaml:1]: Why add these a bunch of places?

**Review by corang — CHANGES_REQUESTED** [review]
So my understanding of the status of this is that we can't get clustering to work while the pods have istio sidecars even with messing with mtls and destinationrules?

File: chart/Chart.yaml [hunk 3]
```diff
@@ -15,4 +15,4 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.0
```
[the remark this answers is not in the document]
Comment (JoeHCQ1) [reply chart/Chart.yaml:18]: OK

File: src/namespace/confluence-ns.yaml (hunk 2 above)
Comment (JoeHCQ1) [reply src/namespace/confluence-ns.yaml:1]: I still tend to add those so if ever concatenated the yaml files remain valid. I think these in particular may have come up when I ran yamllint locally.

File: docs/clustering.md (hunk 1 above)
Comment (JoeHCQ1) [reply docs/clustering.md:41]: Is the comment on line 41 or line 39?

Re Line 39:  it'd mostly just be useful to future me or you whenever we're asked to make it work with Istio.

Re Line 41:  as my client wants confluence and will be serving a very large user load, we'll have to be ready to provide a clustered Confluence installation of some sort. Unless we tell them to put it totally off-cluster, as things stand we will be providing them Confluence clustered without Istio, and so without encryption in transit.

File: common/zarf.yaml [hunk 4]
```diff
@@ -10,7 +10,7 @@ components:
     charts:
       - name: uds-confluence-config
         namespace: confluence
-        version: 0.1.0
```
[the remark this answers is not in the document]
Comment (JoeHCQ1) [reply common/zarf.yaml:13]: OK - boring ;)

**Comment by JoeHCQ1** [comment]
To throw an update in here: we have evidence the clustering can work, and some pointers on how to get it to work. I don't know if we want to chance the consequences of long-lived branches though by not merging this work in even with clustering in a still broken state.

**Review by andrewg-xyz — APPROVED** [review]
Maybe add a test for clustering?

**Comment by Michael-Kruggel** [comment]
Replying for posterity, testing for clustering would require some level of click-ops

**Pull request merged**